src/symbol/projection.js

```diff
@@ -248,8 +256,20 @@ function placeFirstAndLastGlyph(fontScale: number, glyphOffsetArray: GlyphOffset
     return {first: firstPlacedGlyph, last: lastPlacedGlyph};
 }
 
-function requiresOrientationChange(writingMode, firstPoint, lastPoint, aspectRatio) {
-    if (writingMode === WritingMode.horizontal) {
+// Check in the glCoordinate space, the rough estimation of angle between the text line and the Y axis.
+// If the angle if less or equal to 5 degree, then keep the text glyphs unflipped even if it is required.
+function isInFlipRetainRange(firstPoint, lastPoint, aspectRatio) {
+    const deltaY = lastPoint.y - firstPoint.y;
+    const deltaX = (lastPoint.x - firstPoint.x) * aspectRatio;
+    if (deltaX === 0.0) {
+        return true;
+    }
+    const absTangent = Math.abs(deltaY / deltaX);
```

Comment:
👏 Nice avoidance of a `Math.tan` call for each symbol. 😄  I think this indeed works for both +90 and -90 degree orientations.